src/Microsoft.OData.Core/UriParser/Parsers/UriQueryExpressionParser.cs

```diff
@@ -1219,6 +1234,20 @@ private QueryToken ParseSegment(QueryToken parent)
             return new InnerPathToken(propertyName, parent, null);
         }
 
+        /// <summary>
+        /// Parses a $count segment.
+        /// </summary>
+        /// <param name="parent">The parent of the segment node.</param>
+        /// <returns>The lexical token representing the $count segment.</returns>
+        private QueryToken ParseCountSegment(QueryToken parent)
+        {
+            string identifier = this.lexer.CurrentToken.GetIdentifier();

```

Comment:
>string identifier = this.lexer.CurrentToken.GetIdentifier(); [](http://example.com/codeflow?start=12&length=60)

Do you need to do this? Don't you already know it's "$count"?

Reply:
Fixed